llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java

```diff
@@ -1072,7 +1073,9 @@ public Allocator getAllocator() {
 
     @Override
     public void close() throws IOException {
-      // Noop: orcDataReaderRef is owned by the parent object
+      if (orcReader != null){
```

Comment:
Shouldn't we call cleanupReaders() here?

Reply:
makes sense, let me check

Reply:
it seems that was an intentional change, see HIVE-18933 : disable ORC codec pool for now; remove clone

Reply:
yikes, in this I need to figure out whose parent is who :) ... and close from the parent, because while cluster testing, a socket leak was removed by closing the orc reader, so there is a problem with the current implementation

Reply:
okay, uploaded new commit 
https://github.com/apache/hive/pull/5534/commits/df13c21aced3839aa3b532bdf5159203f4bd7610

tested on cluster, problem solved
looks like the cleanupReaders method is called as expected, so closing the orcReader there works

need to note that there were 2 different stacks where the leakage happened, @harishjp  is attaching them to jira as we speak